src/sighash.rs

```diff
@@ -326,6 +326,62 @@ impl<R: Deref<Target = Transaction>> SigHashCache<R> {
     }
 }
 
+/// Hashtype of an input's signature, encoded in the last byte of the signature
+/// Fixed values so they can be casted as integer types for encoding
+#[derive(Copy, Clone, PartialEq, Eq, PartialOrd, Ord, Hash, Debug)]
+pub enum SchnorrSigHashType {
+    /// 0x0: Used when not explicitly specified, defaulting to [`SchnorrSigHashType::All`]
+    Default = 0x00,
+    /// 0x1: Sign all outputs
+    All = 0x01,
+    /// 0x2: Sign no outputs --- anyone can choose the destination
+    None = 0x02,
+    /// 0x3: Sign the output whose index matches this input's index. If none exists,
+    /// sign the hash `0000000000000000000000000000000000000000000000000000000000000001`.
+    /// (This rule is probably an unintentional C++ism, but it's consensus so we have
+    /// to follow it.)
```

Comment:
In 58390807592dbf293343a53d0a125141f103ba40:

This comment is wrong. (See https://github.com/rust-bitcoin/rust-bitcoin/pull/628#discussion_r709361144 where I made the same complaint upstream, but said "not gonna hold up the PR for comment nits" and then we forgot about it).